src/transformers/pipelines.py

```diff
@@ -168,6 +168,24 @@ def __call__(self, *args, **kwargs):
         else:
             return DefaultArgumentHandler.handle_args(args)
 
+class NLIForZeroShotArgumentHandler(ArgumentHandler):
+    """
+    Handles arguments for zero-shot for text classification by turning each possible label into an NLI
+    premise/hypothesis pair.
+    """
+
+    def __call__(self, sequences, labels, hypothesis_template):
+        if isinstance(sequences, str):
+            sequences = [sequences]
+        if isinstance(labels, str):
+            labels = [labels]
+        
+        sequence_pairs = []
+        for sequence in sequences:
+            sequence_pairs += [[sequence, hypothesis_template.format(label)] for label in labels]
```

Comment:
`sequence_pairs.extend([[sequence, hypothesis_template.format(label)] for label in labels])` uses less memory, at least that's what people have been telling me for years